pkg/operator/ceph/version/version.go

```diff
@@ -260,6 +283,14 @@ func IsSuperior(a, b CephVersion) bool {
 				if a.Build > b.Build {
 					return true
 				}
+				if a.Build == b.Build {
+					if a.BuildSuffix != b.BuildSuffix {
+						return true
+					}
+				}
+				if a.Image != b.Image {
```

Comment:
If the image changed we don't know if it's superior, right? We only know it's different. For superiority I think we don't need to know at a depth past `a.Build` so we could revert the changes to this method.